tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -751,4 +765,92 @@ public function theLastResponseShouldBeUnauthorizedReponse(): void {
 			. "\nExpected unauthorized message but got '" . $errorText . "'"
 		);
 	}
+
+	/**
+	 * @param string $user
+	 *
+	 * @return ResponseInterface
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function retrieveUserInformationUsingGraphApi(
+		string $user
+	):ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		return GraphHelper::getUserInformation(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials["username"],
+			$credentials["password"],
+		);
+	}
+
+	/**
+	 * @When /^the user "([^"]*)" retrives (:?her|his) information using the Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws JsonException
+	 */
+	public function userRetrievesHisorHerInformationOfUserUsingGraphApi(
+		string $user
+	):void {
+		$response = $this->retrieveUserInformationUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @Then /^the api response should contains the following information:$/
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theApiResponseForUserShouldContainsTheFollowingInformation(TableNode $table): void {
+		$rows = $table->getRowsHash();
+		$apiResponse = \json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
```

Comment:
we already have function for this
```suggestion
		$apiResponse = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())
```